Assets/Scripts/GameplayObjects/Obstacles/ObstacleController.cs

```diff
@@ -0,0 +1,104 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class ObstacleController : MonoBehaviour
+{
+    #region Editor
+
+    [Range(0, 100)]
+    [SerializeField] private float _obstacleProbability = 15;
+    [SerializeField] private float _obstacleHeight = 0.35f;
+
+    #endregion
+
+    #region Private Variables
+
+    private List<BaseObstacle> _activeObstacleObjects = new List<BaseObstacle>();
+    //private Vector3 _obstacleSpawnZone;
+    private float _spawnZoneMinX, _spawnZoneMaxX, _spawnZoneMinZ, _spawnZoneMaxZ;
+    private int _lastRoadIndex;
+
+    #endregion
+
+    #region Methods
+
+    public void CreateLevelObstacles()
+    {
+        InitializeVariables();
+
+        for (int i = 0; i < GameplayElements.Instance.RoadLength; i++)
+        {
+            var newObs = RandomizeNewObstacle();
+            if(newObs != null)
+            {
+                var pos = RandomPositionOnRoad();
+                GameplayElements.Instance.AddObstacleToRoad(newObs, i, pos);
+            }
+        }
+    }
+
+    public void PullObstacle()
+    {
+        GameplayElements.Instance.GameplayObjectPool.AddObjectToPool(_activeObstacleObjects[0].GetComponent<BaseObstacle>());
+        _activeObstacleObjects.RemoveAt(0);
+    }
+
+    public void RoadPulled(bool hadObstacle)
+    {
+        if(hadObstacle)
+            PullObstacle();
+        var newObs = RandomizeNewObstacle();
+        if (newObs != null)
+        {
+            var pos = RandomPositionOnRoad();
+            GameplayElements.Instance.AddObstacleToRoad(newObs, _lastRoadIndex, pos);
+        }
```

Comment:
this is a duplicate of lines 32-37. simply, you can send the parameters of AddObstacleToRoad to the RandomizeNewObstacle method, if it does create an obstacle it will use them, won't otherwise